traversal/common/Modifiers.java

```diff
@@ -147,12 +148,10 @@ public List<Identifier.Variable.Retrievable> variables() {
             return variables;
         }
 
-        /**
-         * Note: everything that is not explicitly provided with a sort order, is defaulted to Ascending
-         */
-        public Order order(Identifier id) {
-            if (!id.isRetrievable() || !ascending.containsKey(id.asVariable().asRetrievable())) return ASC;
-            else return ascending.get(id.asVariable().asRetrievable());
+
+        public Optional<Order> order(Identifier id) {
```

Comment:
we don't assume the default sort order is `ASC` anymore, which is a domain leak - the user should default to their preferred ordering if the optional is empty